File: src/rectangle_area_service/src/py/ras_server.py
#!/usr/bin/env python
import rospy
from rospy.exceptions import ROSInterruptException
from rectangle_area_service.srv import RectangleAreaService, RectangleAreaServiceRequest, RectangleAreaServiceResponse

def handle_rectangle_area(req):
	rospy.loginfo(req)
	return RectangleAreaServiceResponse(req.width * req.height)

def node():
	rospy.init_node("ras_server", anonymous=False)

	service = rospy.Service("rectangle_area", RectangleAreaService, handle_rectangle_area)
	print("Started service /rectangle_area")

	rospy.spin()

if __name__ == '__main__':
	try:
		node()
	except ROSInterruptException:
		pass

File: src/rectangle_area_service/src/cpp/ras_server.cpp
#include <ros/ros.h>

#include "rectangle_area_service/RectangleAreaService.h"

bool callback(rectangle_area_service::RectangleAreaService::Request& req,
              rectangle_area_service::RectangleAreaService::Response& res) {
  res.area = req.width * req.height;
  return true;
}

int main(int argc, char* argv[]) {
  ros::init(argc, argv, "ras_server");

  ros::NodeHandle node;
  ros::ServiceServer server = node.advertiseService("rectangle_area", callback);
  ROS_INFO("Started service /rectangle_area");

  ros::spin();

  return 0;
}
